Sources/StructuralCore/Structural.swift

```diff
@@ -25,57 +27,55 @@ public protocol Structural {
 }
 
 /// Structural representation of a Swift struct.
-public struct StructuralStruct<Properties> {
-    public var type: Any.Type?
+public struct StructuralStruct<BaseType, Properties> {
+    public var type: BaseType.Type?
     public var properties: Properties
 
     public init(_ properties: Properties) {
         self.type = nil
         self.properties = properties
     }
 
-    public init(_ type: Any.Type, _ properties: Properties) {
+    public init(_ type: BaseType.Type, _ properties: Properties) {
         self.type = type
         self.properties = properties
     }
 }
 
 /// Structural representation of a Swift property.
-public struct StructuralProperty<Value> {
+public struct StructuralProperty<BaseType, Value, WrappedValue> {
+    public var keyPath: KeyPath<BaseType, Value>?
     public var name: String
-    public var value: Value
-    public var isMutable: Bool
+    public var value: WrappedValue
 
-    public init(_ value: Value) {
-        self.name = ""
-        self.value = value
-        self.isMutable = false
+    public var isMutable: Bool {
+        return keyPath is WritableKeyPath<BaseType, Value>
```

Comment:
What happens if `keyPath` is `nil`?

Reply:
It's considered to be not mutable. This happens primarily for associated values in enums, which don't have key paths.

Reply:
Ah, so Swift's special support for `Optional<T>` is kicking in and implicitly promoting the `is WritableKeyPath<BaseType, Value>` to `is WritableKeyPath<BaseType, Value>?`. Is that what's happening? Or is something else going on?

Reply:
Yep, nil values are not considered to be an instance of WritableKeyPath, this seems to work anywhere:
```
  1> let x: Int? = nil
x: Int? = nil
  2> x is Int
$R0: Bool = false
  3> let y: Int? = 1
y: Int? = 1
  4> y is Int
$R1: Bool = true
```

Reply:
Thanks!